src/quo2/components/token_tag.cljs

```diff
@@ -0,0 +1,50 @@
+            [quo2.components.tag :as tag]))
+
+(defn get-value-from-size [size big-option small-option]
+  (if  (= size :big) big-option small-option))
+
+(def icon-container-styles {:display :flex
+                            :align-items :center
+                            :justify-content :center
+                            :position :absolute
+                            :border-radius 100
+                            :margin-left 2})
+
+(defn token-tag
+  "[token-tag opts \"label\"]
+   opts
+   {
+    :token string
+    :value string
+    :size :small/:big
+    :token-img-src :token-img-src
+    :border-color :color
+    :is-required true/false
+    :is-purchasable true/false
+    }"
+  [_ _]
+  (fn [{:keys [token value size token-img-src border-color is-required is-purchasable]
+        :or {size :small border-color (if (=  (theme/get-theme) :dark) colors/purple-60 colors/purple-50)}}]
+
+    [tag/tag {:size size
+              :token-img-src token-img-src
+              :border-color (when is-required border-color)
+              :text-value (str value " " token)
+              :overlay
+              (when (or is-required is-purchasable) [rn/view {:style (merge icon-container-styles {:width 14
+                                                                                                   :height 14
+                                                                                                   :background-color border-color
+                                                                                                   :border-color (if (=  (theme/get-theme) :dark) colors/black colors/white)
+                                                                                                   :border-width 1
+                                                                                                   :right  (get-value-from-size size -3 -5)
+                                                                                                   :bottom (get-value-from-size size (- 32 7 (if is-required 5 4)) (- 24 7 (if is-required 4 2))) ; (- height (icon-height/2) spacing)
+                                                                                                   })}
+                                                     [icons/icon (if is-required :main-icons2/required-checkmark12 :main-icons2/purchasable12) (merge {:no-color true
+                                                                                                                                                       :width (if is-required 7 12)
+                                                                                                                                                       :height (if is-required 6 12)}
+                                                                                                                                                      (when is-required {:container-style {:margin-right 1}}))]])}]))
```

Comment:
better to move `[rn/view ...` to new line. (When it is possible, please, keep lines' length shorter/equal 80 chars.)

Reply:
noted, thanks Parvesh :)